pkg/services/ngalert/eval/eval.go

```diff
@@ -72,6 +72,11 @@ type Result struct {
 	// as EvalMatches (from "classic condition"), and in the future from operations
 	// like SSE "math".
 	EvaluationString string
+
+	// Values contains the RefID and value of reduce and math expressions.
+	// It does not contain values for classic conditions as the values
+	// in classic conditions do not have a RefID.
+	Values map[string]*float64
```

Comment:
@davidmparrott could this be problematic for how we cache evaluations? I think its all cached under the Org ID and the Rule ID, neither of which would be affected given this change.